source/slang/slang-ir-specialize-resources.cpp

```diff
@@ -67,7 +67,10 @@ struct ResourceParameterSpecializationCondition : FunctionCallSpecializeConditio
             else
                 return isIllegalGLSLParameterType(type);
         }
-
+        else if (targetRequest->getTarget() == CodeGenTarget::WGSL)
```

Comment:
We need a `isWGPUTarget()` that takes account for all variants like WGSL_SPIRV_ASM, WGSL_SPIRV etc.